app/services/adventure_service.rb

```diff
@@ -0,0 +1,22 @@
+class AdventureService
+  def self.lock_all_adventure_missions(adventure)
+    adventure_missions = adventure.adventure_missions
+    if adventure_missions.length > 0
```

Comment:
This whole clause can be replaced by 
`adventure_missions.update_all(locked: true) unless adventure_missions.empty?`